devicetypes/vision-kuowei/vision-in-wall-2relays-switch.src/vision-in-wall-2relays-switch.groovy

```diff
@@ -0,0 +1,281 @@
+		encap(endpointNumber, zwave.switchBinaryV1.switchBinarySet(switchValue: 0x00)),
+		encap(endpointNumber, zwave.switchBinaryV1.switchBinaryGet())
+	])
+}
+
+// PING is used by Device-Watch in attempt to reach the Device
+def ping() {
+	refresh()
+}
+
+def refresh() {
+	[encap(1, zwave.switchBinaryV1.switchBinaryGet()), encap(2, zwave.switchBinaryV1.switchBinaryGet())]
+}
+
+// sendCommand is called by endpoint 2 child device handler
+def sendCommand(endpointDevice, commands) {
+	def endpointNumber = 2
+	def result
+	if (commands instanceof String) {
+		commands = commands.split(',') as List
+	}
+	result = commands.collect { encap(endpointNumber, it) }
+	sendHubCommand(result, 100)
+}
+
+def encap(endpointNumber, cmd) {
+	if (endpointNumber == 0) {
+		zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+	}
+	else if (cmd instanceof physicalgraph.zwave.Command) {
+		def cmdTemp = zwave.multiChannelV3.multiChannelCmdEncap(sourceEndPoint: 0x01, destinationEndPoint: endpointNumber).encapsulate(cmd)
+		zwave.securityV1.securityMessageEncapsulation().encapsulate(cmdTemp).format()
+	} else {
+		cmd.format()
+	} 
+}
+
+private getParameterMap() {[[
+		name: "Input switch type (Default: Toggle Switch)", key: "inputSwitchType", type: "enum",
+		parameterNumber: 1, size: 1, defaultValue: 0,
+		values: [
+				0: "Toggle Switch",
+				1: "Momentary Switch",
+		],
+		description: "This item can select input switch type."
+	]
+]}
+
+private syncConfiguration() {
+	def commands = []
+	parameterMap.each {
+		try {
+			if (state.currentPreferencesState."$it.key".status == "syncPending") {
+				commands << encap(0, zwave.configurationV1.configurationSet(scaledConfigurationValue: getCommandValue(it), parameterNumber: it.parameterNumber, size: it.size))
+				commands << "delay 300"			
+				commands << encap(0, zwave.configurationV1.configurationGet(parameterNumber: it.parameterNumber))
+		    }
+		} catch (e) {
+			log.warn "There's been an issue with preference: ${it.key}"
+		}
+	}
+	sendHubCommand(commands, 100)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv1.ConfigurationReport cmd) {
+	log.debug "Configuration report: ${cmd}"
+	def preference = parameterMap.find({it.parameterNumber == cmd.parameterNumber})
+	def key = preference.key
+	def preferenceValue = getPreferenceValue(preference, cmd.scaledConfigurationValue)
+	if (settings."$key" == preferenceValue) {
+		state.currentPreferencesState."$key".value = settings."$key"
+		state.currentPreferencesState."$key".status = "synced"
+	} else {
+		state.currentPreferencesState."$key"?.status = "syncPending"
+		runIn(5, "syncConfiguration", [overwrite: true])
+	}
+}
+
+private getPreferenceValue(preference, value = "default") {
+	def integerValue = value == "default" ? preference.defaultValue : value.intValue()
+	switch (preference.type) {
+		case "enum":
+			return String.valueOf(integerValue)
+		case "boolean":
+			return String.valueOf(preference.optionActive == integerValue)
```

Comment:
since your parameters do not contain anything with type `"boolean"` the code referencing it here and elsewhere can be removed

Reply:
Already removed type "boolean" in all parameters function.